hydroflow/src/compiled/pull/symmetric_hash_join.rs

```diff
@@ -88,6 +88,14 @@ where
     type Item = (Key, (V1, V2));
 
     fn next(&mut self) -> Option<Self::Item> {
+        if self.state.0.table.is_empty() && self.lhs.peek().is_none() {
```

Comment:
0 is the lhs and 1 is the rhs, should this be rhs.peek and the one below be lhs.peek()?

Reply:
The idea of this condition is it expresses whether the table will _remain_ empty. So we are saying that the table currently has no values and there are no values that can be added to it.

Reply:
I understand what it's doing now.